Libraries/LibJS/Runtime/StringPrototype.cpp

```diff
@@ -185,6 +186,77 @@ JS_DEFINE_NATIVE_FUNCTION(StringPrototype::starts_with)
     return Value(string.substring(start, search_string_length) == search_string);
 }
 
+static bool is_regexp(VM& vm, GlobalObject& global_object, Value possible_regexp)
+{
+    if (!possible_regexp.is_object())
+        return false;
+
+    auto matcher = possible_regexp.as_object().get(vm.well_known_symbol_match());
+    if (vm.exception())
+        return false;
+    if (!matcher.is_empty() && !matcher.is_undefined()) {
+        return !matcher.to_boolean();
+    }
+
+    auto regexp_as_object = possible_regexp.to_object(global_object);
+    if (vm.exception())
+        return false;
+    if (regexp_as_object->is_regexp_object()) {
+        return true;
+    }
+
+    return false;
+}
+
+JS_DEFINE_NATIVE_FUNCTION(StringPrototype::ends_with)
+{
+    auto string = ak_string_from(vm, global_object);
+    if (string.is_null())
+        return {};
+
+    auto search_string_value = vm.argument(0);
+    if (vm.exception())
+        return {};
```

Comment:
No need for an exception check here :)

```suggestion
    auto search_string_value = vm.argument(0);
```